pkg/apis/druid/v1alpha1/druid_types.go

```diff
@@ -229,6 +229,11 @@ type DruidNodeSpec struct {
 	// Optional: Ingress Spec
 	Ingress *extensions.IngressSpec `json:"ingress,omitempty"`
 
+	// Optional: Persitance Volume Spec for Deployments
+	PersistentVolumeClaim *v1.PersistentVolumeClaimSpec `json:"persistentVolumeClaim,omitempty"`
+
+	// Optional: MounthPath to mount volume
+	PersistentVolumeClaimMounthPath string `json:"persistentVolumeClaimMountPath,omitempty"`
```

Comment:
I am slightly concerned over existence of these two and volumeXXX keys that we already had for StatefulSet. I haven't explored in detail but it would be good to somehow, at spec level, have only one set of keys regarding the persistent volumes and that be used for both StatefulSet and Deployment.

Reply:
i had the same query :) , since both have  ```type persistentVolumeClaim```,
 The reason i went was just to not have confusion and follow as  per k8s convention deployments have pvc mounts and sts have volumeclaimtemplates. In case you want to have a common name, it should not conflict with k8s convention. 
open to suggestions :)